### ISSUE TITLE
Should deleteForward change the direction of model#deleteContent() merge blocks?

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5#3115]
Let's see the two following cases:

TC1: `<h1>Foo</h1><p>[]Bar</p>` + delete (backspace)
TC2: `<h1>Foo[]</h1><p>Bar</p>` + forward delete

Now, we could handle these cases differently. Right now, in both of them the result will be `<h1>Foo[]Bar</h1>`. However, perhaps the user's intention is a bit different when pressing the backward and forward delete keys.

When using the editor I'm expecting that forward delete will merge the current block "to the right", so it will keep that block (`<p>Foo[]Bar</p>`). We should perhaps check some other editors to see if they handle forward delete differently.

**Comment by zehawki** [comment]
Thats a complex issue name and I for one would not have matched it to what I reported a while ago on #6282 :-)

I'm just copy pasting from that issue for easy reference:

The para text takes the formatting of the previous line even though it was deleted.

Here's a gif showing whats going on:

![captured](https://user-images.githubusercontent.com/8095144/74720544-fd190b80-525b-11ea-85fe-e3b6be628fbb.gif)

I looked into the HTML as its being generated and see the issue:

After the line is deleted, the formatting tags are not removed along with the line, so the HTML contains:
`<h2>&nbsp;</h2><p>Like all the great things on earth traveling teaches us by example. Here are some of the most precious lessons I’ve learned over the years of traveling.</p>`
And then when delete is pressed, the next para text takes the h2 tag.

This is a very unintuitive behavior and something I've not seen in any other editor. To cross check I performed exactly the same in Word, Google docs, Quill and clearly they handle this as expected.

Froala does the same thing as CK, BUT it has a saving grace - when the heading line is deleted and another backspace is pressed, the line formatting disappears.

I strongly feel this behavior should be changed since its quite unexpected & puzzling for the end users (who reported this issue to me) and I find it equally puzzling :-)
